File: Angular/src/app/inventory/interfaces/inventory-note.ts
import {InventoryNoteDetail} from './inventory-note-detail';
import {Contact} from '../../contact/interfaces';
import {Warehouse} from '../../system/interfaces';

export class InventoryNote {
  constructor(
    public id: number | any = null,
    public contact: Contact | any = undefined,
    public contactId: number = 0, // Id de Contacto.
    public warehouse: Warehouse | any = undefined,
    public warehouseId: number = 0, // Id Almacén.
    public noteType: string = '', // Tipo de Nota (Ingreso|Salida).
    public motivo: string = '', // Motivo de Inventario.
    public startDate: any = null, // Fecha de Registro.
    public status: string = '', // Estado de la Nota.
    public inventoryNoteDetails: Array<InventoryNoteDetail> = new Array<InventoryNoteDetail>()) {
  }
}

File: src/Nebula/Data/Models/InventoryNote.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Nebula.Data.Models;

// TODO: refactoring.
public class InventoryNote
{
    public int Id { get; set; }

    /// <summary>
    /// Id de Contacto.
    /// </summary>
    public int? ContactId { get; set; }

    /// <summary>
    /// Clave foránea Contacto.
    /// </summary>
    [ForeignKey("ContactId")]
    public Contact Contact { get; set; }

    /// <summary>
    /// Id Almacén.
    /// </summary>
    public int? WarehouseId { get; set; }

    /// <summary>
    /// Clave foránea Almacén.
    /// </summary>
    [ForeignKey("WarehouseId")]
    public Warehouse Warehouse { get; set; }

    /// <summary>
    /// Tipo de Nota (Ingreso|Salida).
    /// </summary>
    [MaxLength(250)]
    public string NoteType { get; set; }

    /// <summary>
    /// Motivo de Inventario.
    /// </summary>
    [MaxLength(250)]
    public string Motivo { get; set; }

    /// <summary>
    /// Fecha de Registro.
    /// </summary>
    [DataType(DataType.Date)]
    public DateTime StartDate { get; set; }

    /// <summary>
    /// Estado de la Nota.
    /// </summary>
    [MaxLength(250)]
    public string Status { get; set; }

    /// <summary>
    /// Año de Inventario.
    /// </summary>
    [MaxLength(250)]
    public string Year { get; set; }

    /// <summary>
    /// Mes de Inventario.
    /// </summary>
    [MaxLength(250)]
    public string Month { get; set; }

    /// <summary>
    /// Detalles Nota de Inventario.
    /// </summary>
    public List<InventoryNoteDetail> InventoryNoteDetails { get; set; }
}
